tests/research/test_b2_05_flow_absorption.py

```
@@ -0,0 +1,1035 @@
+    assert gates["parameter_robustness"] is True
+    assert gates["year_stability"] is True
+    assert len(neighborhoods) == 1
+    assert neighborhoods[0]["H_pair"] == [30, 60]
+    assert neighborhoods[0]["W"] == [15, 30]
+
+
+def test_year_stability_three_of_five_fails_four_of_five_passes():
+    cell = {
+        "W": 15,
+        "H": 30,
+        "per_cell_conditions": {name: True for name in lib.PER_CELL_GATE_NAMES},
```

Ruff (0.16.3)

[warning] Unnecessary dict comprehension for iterable; use `dict.fromkeys` instead

Replace with `dict.fromkeys(iterable)`)

(C420)